pkg/login/social/generic_oauth.go

```diff
@@ -232,13 +221,16 @@ func (s *SocialGenericOAuth) extractEmail(data *UserInfoJson) string {
 }
 
 func (s *SocialGenericOAuth) extractRole(data *UserInfoJson) string {
-	if s.roleAttributePath != "" {
-		role := s.searchJSONForAttr(s.roleAttributePath, data.rawJSON)
-		if role != "" {
-			return role
-		}
+	if s.roleAttributePath == "" {
+		return ""
 	}
-	return ""
+
+	role, err := s.searchJSONForAttr(s.roleAttributePath, data.rawJSON)
+	if err != nil {
+		s.log.Error("Failed to extract role", "error", err)
```

Comment:
I think it would be better to modify `extractRole` into returning also an error, and returning this error here.

@marefr What do you think?